ts/src/flexible-event/privacy.ts

```diff
@@ -214,3 +214,43 @@ function epsilonToBoundInfoGainAndDp(
     return epsilon
   }
 }
+
+// Returns the effective epsilon needed to satisfy an information gain bound
+// given a number of output states in the q-ary symmetric channel.
+export function epsilonToBoundInfoGainAndDp(
+  numStates: number,
+  infoGainUpperBound: number,
+  epsilonUpperBound: number
+): number {
+  // Calculate epsilon to double precision.
+  const candidate: number[] = new Array<number>(64).fill(0)
```

Comment:
Instead of using an array of 64 numbers, can we use typed arrays operating on a shared buffer that lets us perform bitwise ops?

```js
const buf = new ArrayBuffer(8)
const floatView = new Float64Array(buf)
const intView = new BigUint64Array(buf)
for (let byte = 0; byte < 8; ++byte) {
  for (let bit = 0n ; bit < 8n; ++bit) {
    intView[byte] |= 1n << bit
    const epsilon = floatView[0]!
    ...
    intView[byte] &= ~(1n << bit)
```

Reply:
We may need to use [`DataView`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) to guarantee endianness, since I assume we need big-endian for the search to hold.

Reply:
```js
  const buf = new ArrayBuffer(8)
  const dataView = new DataView(buf)

  for (let bit = 0n; bit < 64n; ++bit) {
    dataView.setBigUint64(0, dataView.getBigUint64(0) | (1n << bit))

    const epsilon = dataView.getFloat64(0)
    if (epsilon > epsilonUpperBound) {
      dataView.setBigUint64(0, dataView.getBigUint64(0) & ~(1n << bit))
      continue
    }

    const infoGain = maxInformationGain(numStates, epsilon)

    if (infoGain > infoGainUpperBound) {
      dataView.setBigUint64(0, dataView.getBigUint64(0) & ~(1n << bit))
    } else if (epsilon === epsilonUpperBound) {
      return epsilon
    }
  }

  return dataView.getFloat64(0)
```

Reply:
Ha! I forgot JavaScript could offer this lower level. Will look at it tomorrow. Thanks!

Reply:
This is very helpful, thanks again!